src/crimson/osd/osd.cc

```diff
@@ -608,6 +610,52 @@ seastar::future<> OSD::_send_boot()
   return monc->send_message(std::move(m));
 }
 
+seastar::future<> OSD::_add_device_class()
+{
+  LOG_PREFIX(OSD::_add_device_class);
+  if (!local_conf().get_val<bool>("osd_class_update_on_start")) {
+    co_return;
+  }
+
+  auto get_device_class = [FNAME, this]() -> seastar::future<std::string> {
+    try {
+      std::string device_class = co_await store.get_default_device_class();
+      co_return device_class;
+    } catch (const std::exception& e) {
+      ERROR("Failed to get device class: ", e.what());
+      co_return "";
+    }
+  };
```

Comment:
There's no need for try/catch we discarding exceptions as we handle error with Crimson's errorator concept.
See `TEST_F(coroutine_test_t, test_ertr_coroutine_error)`

Reply:
We can skip the `get_device_class` wrapper: 
```
std::string device_class = co_await store.get_default_device_class();
```

Reply:
done